src/modules/sensors/sensors.cpp

```diff
@@ -507,7 +509,7 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 							selected_source = b;
 
 							/* Move the selected_source to instance 0 */
-							if (_battery_pub_intance0ndx != b) {
+							if (BOARD_NUMBER_BRICKS > 1 && _battery_pub_intance0ndx != b) {
```

Comment:
@jlecoeur - Thank you for the details. 

The irony is that once #7618 comes in. We may need to refactor this or make the retrieval  # bricks, and details a runtime decision. This is need so that on V3 HW it looks like dual brick (not Servo power).  But on V2 it would return 1 and indicate (Servo power) , V3 would return 2.

We need to insure that `selected_source`  [is set](https://github.com/jlecoeur/Firmware/blob/7a061ebab9ef383fc33b8c0d8e4c198060238065/src/modules/sensors/sensors.cpp#L547), and there is lazy initialization on the pointers so it is passive, until the 
subscription pointers are set. 

Since the fix is effectively an #if BOARD_NUMBER_BRICKS > 1, should we wrap the block that way and add a comment? 

 If we change it to with #if BOARD_NUMBER_BRICKS == 1 and 
```
#if BOARD_NUMBER_BRICKS == 1 
   selected_source  = 0;
#else 
Code as is
#ifdef 
```
>I tried to move the initialization of the variable to the cpp file, but that did not help the compiler. I left it there because this is done like this elsewhere, but I can revert if you prefer.

Let's let @dagar chime in on that one.